catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/resource/download/ReliableResourceDownloadManagerTest.java

```diff
@@ -652,6 +667,88 @@ public void testStreamToClientExceptionDuringProductDownloadCachingEnabled() thr
         cleanup();
     }
 
+    @Test
+    public void testGetDownloadsInProgress() {
+        List<String> downloadIds = new ArrayList<>();
+        downloadIds.add("03e3f850-240c-4cc6-a5a3-318dc34ad4bd");
+        downloadIds.add("5tygH67-345t-3er5-r86y-5tyZHU7UGD092");
+        downloadIds.add("9tY75f4-345t-4er8-jj87-Y67hJJK098yaq");
+        downloadIds.add("56lJOJl-45gg-3wf5-ww23-tldf7HewfhweJ");
+        downloadIds.add("rRdlefj-ggal-erty-rr6e-ZZefoeje546kL");
+
+        List<Map<String, String>> downloadStatusMaps = getDownloadStatusMaps(downloadIds);
+
+        DownloadStatusInfo downloadStatusInfo = mock(DownloadStatusInfo.class);
+        when(downloadStatusInfo.getAllDownloads()).thenReturn(downloadIds);
+        when(downloadStatusInfo.getDownloadStatus(downloadIds.get(0)))
+                .thenReturn(downloadStatusMaps.get(0));
+        when(downloadStatusInfo.getDownloadStatus(downloadIds.get(1)))
+                .thenReturn(downloadStatusMaps.get(1));
+        when(downloadStatusInfo.getDownloadStatus(downloadIds.get(2)))
+                .thenReturn(downloadStatusMaps.get(2));
+        when(downloadStatusInfo.getDownloadStatus(downloadIds.get(3)))
+                .thenReturn(downloadStatusMaps.get(3));
+        when(downloadStatusInfo.getDownloadStatus(downloadIds.get(4)))
+                .thenReturn(downloadStatusMaps.get(4));
+
+        ReliableResourceDownloadManager reliableResorceDownloadManager = new ReliableResourceDownloadManager(
+                null, downloadStatusInfo, null);
+
+        List<DownloadInfo> downloadInfoList = reliableResorceDownloadManager
+                .getDownloadsInProgress();
+
+        assertThat(downloadInfoList.size(), is(2));
+        for (DownloadInfo downloadInfo : downloadInfoList) {
+            assertThat(downloadInfo.isDownloadInState(DownloadState.IN_PROGRESS), is(true));
+        }
+    }
+
+    private List<Map<String, String>> getDownloadStatusMaps(List<String> downloadIds) {
+        List<Map<String, String>> downloads = new ArrayList<>();
+        Map<String, String> downloadStatus1 = new HashMap<>();
```

Comment:
Looking at the mess below, we definitely need a helper method for populating the map. Something like: 
```
createDownloadStatusMap(downloadIds.get(0), 
      "image1.jpg", 
      DownloadState.IN_PROGRESS.name(), "862978048",
      "76",
      "admin");
```